### PR TITLE
Disambiguate invalid model name errors

### PR DESCRIPTION
**Pull request by msabramo**
because that error can be thrown in several different places, so knowing the function it's being thrown from can be very useful for debugging.

In fact, we faced this difficulty when @taralika and I were looking into https://github.com/BerriAI/litellm/issues/3372

### CODE REVIEW

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
tests need to be updated before merging this in

**Comment by msabramo** [comment]
> tests need to be updated before merging this in

Updated tests in 77f2e7ccc.

**Comment by ishaan-jaff** [comment]
Good to approve once merge conflicts are resolved @msabramo

**Comment by msabramo** [comment]
> Good to approve once merge conflicts are resolved @msabramo

Rebased and fixed conflicts and now the test update commit is 6d154e231.

**Review by ishaan-jaff — APPROVED** [review]
lgtm !

**Pull request merged**
